src/LanguageDetector.js

```diff
@@ -54,10 +54,10 @@ class LanguageDetector {
 
       let detections = this.detectors[detectorName].lookup(req, res, this.options);
       if(!detections) return;
-      if (typeof detections === 'string') detections = [detections];
+      if (!Array.isArray(detections)) detections = [detections];
 
       detections.forEach(lng => {
-        if (found) return;
+        if (found || typeof lng !== 'string') return;
 
         let cleanedLng = this.services.languageUtils.formatLanguageCode(lng);
```

Comment:
don't the code finally fail here: formatLanguageCode expects a string...if there will get passed an object down that function will fail...

Reply:
Yes - I don't actually want to support querystring objects (such as `?lang[foo]=en&lang[bar]=zh`) - how should that be parsed? I just want to prevent a crash if we see one.

Reply:
But who does call you website with that qs? i mean more or less that is under your control...i don't think users get on the idea to pass that as array by themself...

Reply:
If you're running a website, anyone can call any endpoint with anything. This middleware crashes (synchronous crash) if someone sends a specially-crafted QS. That is bad. We can't control all clients.

This PR fixes the crash and ignores the misformatted value.